### PR TITLE
General improvements.

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Framework/Extensions/Color4Extensions/Color4Extensions.cs [hunk 1]
```diff
@@ -68,5 +68,46 @@ public static Color4 Multiply(Color4 first, Color4 second)
...
+                Math.Min(1, colour.R * (1 + amount)),
+                Math.Min(1, colour.G * (1 + amount)),
+                Math.Min(1, colour.B * (1 + amount)),
+                colour.A);
+        }
+
+        /// <summary>
+        /// Lightens a colour in a way more friendly to dark or strong colours.
+        /// </summary>
+        public static Color4 Lighten2(this Color4 colour, float amount)
```
Comment (Tom94) [inline osu.Framework/Extensions/Color4Extensions/Color4Extensions.cs]: Have you tried just lightening / darkening in linear space using the ToLinear and ToSRGB functions above? Might give you better results.

File: osu.Framework/Graphics/Drawable.cs [hunk 2]
```diff
@@ -714,7 +714,7 @@ public ColourInfo ColourInfo
         /// Colour of this Drawable in sRGB space. Only valid if no individual colours
         /// have been specified for each corner vertex via <see cref="ColourInfo"/>.
         /// </summary>
-        public SRGBColour Colour
+        public virtual SRGBColour Colour
```
Comment (Tom94) [inline osu.Framework/Graphics/Drawable.cs]: Where is the virtual needed? I didn't spot a place in your osu PR.

Comment (peppy) [inline osu.Framework/Graphics/Drawable.cs]: future use in game project. want to be able to (for instance) pass the colour setting on to a deeper child without it affecting the whole container.

Comment (Tom94) [inline osu.Framework/Graphics/Drawable.cs]: I think using a separate property like `WidgetColour` would be better in the example you gave. Using `Colour` for that would be breaking expected behaviour for skins. There is quite some value to have properties which reliably do the same thing.

File: osu.Framework/Extensions/Color4Extensions/Color4Extensions.cs [hunk 3]
```diff
@@ -68,5 +68,46 @@ public static Color4 Multiply(Color4 first, Color4 second)
...
+
+        /// <summary>
+        /// Returns a darkened version of the colour.
+        /// </summary>
+        /// <param name="colour">Original colour</param>
+        /// <param name="amount">Percentage light reduction</param>
+        public static Color4 Darken(this Color4 colour, float amount)
+        {
+            return new Color4(
+                Math.Max(0, colour.R / (1 + amount)),
```
Comment (Tom94) [inline osu.Framework/Extensions/Color4Extensions/Color4Extensions.cs]: Max is unnecessary. Also, can refactor Lighten and Darken to use a common Multiply method that multiplies a Color4 by a scalar (float). Darken can then just use the inverse of the factor Lighten uses.

**Pull request merged**